protocols/mdns/src/behaviour/iface.rs

```diff
@@ -107,7 +110,9 @@ impl InterfaceState {
                 SocketAddr::new(IpAddr::V6(Ipv6Addr::UNSPECIFIED), 0)
             }
         };
-        let send_socket = Async::new(UdpSocket::bind(bind_addr)?)?;
+        let std_socket = UdpSocket::bind(bind_addr)?;
+        let send_socket = U::from_std(std_socket)?;
```

Comment:
For the sake of minimizing the diff, we could write this similar to how the old code did it:

```suggestion
        let send_socket = U::from_std(UdpSocket::bind(bind_addr)?)?;
```